app/controllers/customer_controller.rb

```diff
@@ -0,0 +1,49 @@
+class CustomerController < ApplicationController
+
+  def index
+    @customers = Customer.all
+  end
+
+  def new
+    #display a form that passes customer params to the create route
+  end
+
+  def create
+    @customer = Customer.new(params[:customer])
+    if @customer.save
+      session[:user_id] = @customer.id
+      redirect_to @customer
+    else
+      @errors = @customer.errors.full_messages
+      render :new
+  end
+
+  def show
+    @customer = Customer.find(params[:id])
+  end
+
+  def edit
+    @customer = Customer.find(params[:id])
+  end
+
+  def update
+    @customer = Customer.find(params[:id])
+    if @customer.update(customer_params)
+      redirect_to @user, notice: 'Your profile has been updated!' #profile view,
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @customer = Customer.find(params[:id])
+    @customer.destroy
+  end
+
+  private
+    def customer_params
+      params.require(:customer).permit(:first_name, :last_name, :password, :email, :username)
+    end
+
+end
+
```

Comment:
I think the majority of these routes will be returning Json to the phone view, partucularly becuase the merchant will lnot have access to the user routes and a user will most likely not be accessing their routes through http. Or we can set it up that way so the user can only access their routes through their phone, and we can then dictate that all routes will return json and then we won't have to worry aout rendering html.